steps/index.lab.md

```diff
@@ -314,72 +342,71 @@ Then, edit the `onCreate()` method to initialize Firebase in the same way you di
+
+        startActivityForResult(signInIntent, RC_SIGN_IN)
+    } else {
+        goToMainActivity()
+    }
 }
 ```
 
-Next, implement the `onActivityResult()` method to handle the sign in result. If the result of the Google Sign-In was successful, use the account to authenticate with Firebase:
+Next, implement the `onActivityResult()` method to handle the sign in result. If the result of the signin was successful, continue to `MainActivity`:
 
 **SignInActivity.kt**
 
-```
+```kotlin
 public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
     super.onActivityResult(requestCode, resultCode, data)
 
-    // Result returned from launching the Intent in signIn()
+    // Handle FirebaseUI Authentication result
     if (requestCode == RC_SIGN_IN) {
-        val task = GoogleSignIn.getSignedInAccountFromIntent(data)
-        try {
-            // Google Sign In was successful, authenticate with Firebase
-            val account = task.getResult(ApiException::class.java)
-            firebaseAuthWithGoogle(account)
-        } catch (e: ApiException) {
-            // Google Sign In failed, update UI appropriately
-            Log.w(TAG, "Google sign in failed", e)
-        }
-    }
-}
-```
-
-Implement the `firebaseAuthWithGoogle()` method to authenticate with the signed in Google account:
-
-**SignInActivity.kt**
-
-```
-private fun firebaseAuthWithGoogle(acct: GoogleSignInAccount?) {
-    Log.d(TAG, "firebaseAuthWithGoogle:" + acct?.id)
-    val credential = GoogleAuthProvider.getCredential(acct?.idToken, null)
-    auth.signInWithCredential(credential)
-        .addOnSuccessListener(this) {
-            // If sign in succeeds the auth state listener will be notified and logic to
-            // handle the signed in user can be handled in the listener.
-            Log.d(TAG, "signInWithCredential:success")
-            startActivity(Intent(this@SignInActivity, MainActivity::class.java))
-            finish()
-        }
-        .addOnFailureListener(this) { e -> // If sign in fails, display a message to the user.
-            Log.w(TAG, "signInWithCredential", e)
+        val response = IdpResponse.fromResultIntent(data)
+        if (resultCode == RESULT_OK) {
+            Log.d(TAG, "Sign in successful!")
+            goToMainActivity()
+        } else {
             Toast.makeText(
-                this@SignInActivity, "Authentication failed.",
-                Toast.LENGTH_SHORT
-            ).show()
+                    this,
+                    "There was an error signing in",
+                    Toast.LENGTH_LONG).show()
+            if (response == null) {
+                Log.w(TAG, "Sign in canceled")
+            } else {
+                Log.w(TAG, "Sign in error", response.error)
+            }
         }
+    }
 }
 ```
 
-That's it! You've implemented authentication using Google as an Identity Provider in just a few method calls and without needing to manage any server-side configuration.
```

Comment:
Should we mention in this step that the developer should be able to see the registered users in the Auth Emulator?

Reply:
Yep!